ledger/notary-plugin/src/main/java/net/corda/v5/ledger/notary/plugin/core/NotaryException.java

```diff
@@ -0,0 +1,46 @@
+package net.corda.v5.ledger.notary.plugin.core;
+
+import net.corda.v5.base.annotations.CordaSerializable;
+import net.corda.v5.base.exceptions.CordaRuntimeException;
+import net.corda.v5.crypto.SecureHash;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The base class for all notary errors that can be returned by the notary itself (plugin). Specific notary exceptions
+ * must not inherit from this class directly, they should use one of the inner classes. To create a notary exception use
+ * @see NotaryExceptionFatal
+ * or 
+ * @see NotaryExceptionUnknown
+ */
+@CordaSerializable
+abstract class NotaryException extends CordaRuntimeException {
+
+    @NotNull
+    private final String notaryErrorMessage;
+    @Nullable
+    private final SecureHash txId;
```

Comment:
I don't think you need to annotate private Java fields because the Kotlin compiler is never going to see these.

Reply:
Yeah that's a good point!